### PR TITLE
Use junit in functional tests

### PR DESCRIPTION
**Pull request by balamurugana**

### CODE REVIEW

**Review by ebozduman — CHANGES_REQUESTED** [review]
Functional test run fails...
```
$ ./gradlew runFunctionalTest` also fails with error.
> Task :functional:runFunctionalTest FAILED

FAILURE: Build failed with an exception.
```

File: functional/FunctionalTest.java [hunk 1]
```diff
@@ -2606,27 +2585,29 @@ public static void testGetObjectRetention(SetObjectRetentionArgs args) throws Ex
...
 
     if (expectedDate == null) {
-      if (date != null) {
-        throw new Exception("retention retain-until-date: expected: <null>, got: " + date);
-      }
-    } else if (!date.withNano(0).equals(expectedDate.withNano(0))) {
-      throw new Exception(
+      Assert.assertNull("retention retain-until-date: expected: <null>, got: " + date, date);
+    } else {
+      Assert.assertEquals(
```
[the remark this answers is not in the document]
Comment (ebozduman) [reply functional/FunctionalTest.java:2604]: Bala explained to me that the `Assert.assertNull` actually means "make sure the condition is **NOT** Null".
So, all my comments are wrong. Resolved all of them.

**Review by ebozduman — APPROVED** [review]
LGTM

**Review by kannappanr — APPROVED** [review]
LGTM

**Pull request merged**